src/lib/modules/ModuleContainer/index.js

```diff
@@ -0,0 +1,37 @@
+class ModuleContainer {
+  constructor(io, { modules = [] }) {
+    this.io = io;
+    this.handle = this.handle.bind(this);
+    this.registeredModule = modules;
+    this.activatedModule = [];
+  }
+
+  activate(name) {
```

Comment:
I think activation / deactivation should be added later. It actually create noise on the PR.

Maybe we should divide this "big" one into multiple smaller ? So that we can have little feedback easily ?